### PR TITLE
Add content to kotlin documentation 'toDateTimePeriod'.

### PR DESCRIPTION
**Pull request by DylantheChicken**
~/docs/content/kotlin/ concepts/datetime/terms/toDateTimePeriod/

### Description

**Closes Issue #3865**
Made changes to the following path by adding a folder 'toDateTimePeriod'.
**docs/content/kotlin/ concepts/datetime/terms/**
Added new file to path **~/content/kotlin/ concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md**

### Type of Change
- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR.

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  - 'Days'
+  - 'Hours'
+  - 'Minutes'
+  - 'Seconds'
+  - 'Nanoseconds'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+## Description
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:23]:
```suggestion
```
No need for this heading

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+The **.toDateTimePeriod()** method in Kotlin is a versatile function that transforms a Duration object into a DateTimePeriod representation. This conversion is useful when dealing with durations in Kotlin's date and time API, especially when the durations need to be expressed in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for easier interpretation and manipulation.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:25]:
```suggestion
The **`.toDateTimePeriod()`** method in Kotlin is a versatile function that transforms a Duration object into a DateTimePeriod representation. This conversion is useful when dealing with durations in Kotlin's date and time API, especially when the durations need to be expressed in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for easier interpretation and manipulation.
```

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Syntax
+
+```pseudo
+duration.toDateTimePeriod(): DateTimePeriod
+```
+
++ **duration**: This is a Duration object representing a length of time.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:33]:
```suggestion
- **duration:** This is a Duration object representing a length of time.
```

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+The method returns a DateTimePeriod object, which is a normalized representation of the input duration in terms of larger time units (years, months, days, etc.).
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:35]:
```suggestion
> **Note:** The method returns a DateTimePeriod object, which is a normalized representation of the input duration in terms of larger time units (years, months, days, etc.).
```

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    // Optionally, adding some extra time for breaks (let's say 12 hours)
+    val totalWithDurationBreaks = totalDuration + Duration.hours(12)
+
+    // Converting the total duration with breaks to a DateTimePeriod
+    val dateTimePeriod = totalWithDurationBreaks.toDateTimePeriod()
+
+    println("Total Duration in DateTimePeriod: $dateTimePeriod")
+}
+```
+The expected output should be similar to the following:
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:61]:
```suggestion

The expected output should be similar to the following:

```
Add white spaces

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    // Summing up all durations
+    val totalDuration = designingDuration + codingDuration + testingDuration
+
+    // Optionally, adding some extra time for breaks (let's say 12 hours)
+    val totalWithDurationBreaks = totalDuration + Duration.hours(12)
+
+    // Converting the total duration with breaks to a DateTimePeriod
+    val dateTimePeriod = totalWithDurationBreaks.toDateTimePeriod()
+
+    println("Total Duration in DateTimePeriod: $dateTimePeriod")
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:58]:
```suggestion
fun main() {
    // Durations for different activities in hours
    val designingDuration = Duration.hours(31)  
    val codingDuration = Duration.hours(57) 
    val testingDuration = Duration.hours(14)    

    // Summing up all durations
    val totalDuration = designingDuration + codingDuration + testingDuration

    // Optionally, adding some extra time for breaks 
    val totalWithDurationBreaks = totalDuration + Duration.hours(12)

    // Converting the total duration with breaks to a DateTimePeriod
    val dateTimePeriod = totalWithDurationBreaks.toDateTimePeriod()

    println("Total Duration in DateTimePeriod: $dateTimePeriod")
```
Remove the redundant comments

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
+---
+Title: '.toDateTimePeriod()'
+Description: 'Interprets an ISO-8601 formatted duration string, converting it into a DateTimePeriod object. In cases where the string's time components are either missing or amount to zero, the function yields a DatePeriod instead.'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:3]: Make this short and descriptive. This long explanation is supposed to be done after the meta data not here.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have few comment for you @DylantheChicken

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+Subjects:
+  - 'Computer Science'
+  - 'Software Engineering'
+  - 'Other comparable degrees'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:7]: Add the subjects here which are only present [here](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md)

Comment (DylantheChicken) [reply content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:7]: Thank you for this! I thought it was somewhere, did not know where to find it.

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md (hunk 7 above)
Comment (DylantheChicken) [reply content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:3]: Noted with thanks!

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md (hunk 8 above)
Comment (dakshdeepHERE) [reply content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:7]: It's okay @DylantheChicken ✅

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+The expected output should be similar to the following:
+
+```mathematica
+Total Duration in DateTimePeriod: DateTimePeriod(days = 5, hours = 4)
+```
+
+## Exception Throws
+**IllegalArgumentException** - if the total number of months in hours, minutes, seconds and nanoseconds overflows a Long.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:68]:
```suggestion
>**Note:** `Exception Throws (IllegalArgumentException)-` if the total number of months in hours, minutes, seconds, and nanoseconds overflows a Long.
```
Let's remove this heading and add a note here.

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  - 'Functions'
+  - 'Data Structures'
+  - 'DatePeriod'
+  - 'Years'
+  - 'Months'
+  - 'Days'
+  - 'Hours'
+  - 'Minutes'
+  - 'Seconds'
+  - 'Nanoseconds'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:18]:
```suggestion
```
No need for these tags

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+Title: '.toDateTimePeriod()'
+Description: 'Interprets an ISO-8601 formatted duration string, converting it into a DateTimePeriod object. In cases where the string's time components are either missing or amount to zero, the function yields a DatePeriod instead.'
+Subjects:
+  - 'Computer Science'
+  - 'Software Engineering'
+  - 'Other comparable degrees'
+Tags:
+  - 'Functions'
+  - 'Data Structures'
+  - 'DatePeriod'
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:11]: This is not present [here](https://github.com/Codecademy/docs/blob/main/documentation/tags.md). Either add `DatePeriod` in `tags.md` file or use the tag `Date` instead.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few more changes

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  - 'Methods'
+  - 'Data Structures'
+  - 'Date'
+  - 'Time'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.toDateTimePeriod()`** method in Kotlin is a versatile function that transforms a Duration object into a DateTimePeriod representation. This conversion is useful when dealing with durations in Kotlin's date and time API, especially when the durations need to be expressed in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for easier interpretation and manipulation.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:17]:
```suggestion
The **`.toDateTimePeriod()`** method in Kotlin is a versatile function that transforms a Duration object into a [DateTimePeriod](https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/-date-time-period/#:~:text=The%20date%20components%20are%3A%20years,periodUntil).) representation. This conversion is particularly useful when working with durations in Kotlin's date and time API, especially when expressing durations in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for easier interpretation and manipulation.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
One change

**Comment by DylantheChicken** [comment]
> One change

This has been committed. Thanks for all these pointers!

**Comment by dakshdeepHERE** [comment]
@DylantheChicken  lets move to second review.

**Comment by ishg-153** [comment]
@dakshdeepHERE Can you approve the changes while I work on the second review?

**Review by dakshdeepHERE — APPROVED** [review]

**Comment by dakshdeepHERE** [comment]
@ishg-153 done 👍

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+The **`.toDateTimePeriod()`** method in Kotlin is a versatile function that transforms a Duration object into a [DateTimePeriod](https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/-date-time-period/#:~:text=The%20date%20components%20are%3A%20years,periodUntil).) representation. This conversion is particularly useful when working with durations in Kotlin's date and time API, especially when expressing durations in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for easier interpretation and manipulation.
+
+## Syntax
+
+```pseudo
+duration.toDateTimePeriod(): DateTimePeriod
+```
+
+- **duration:** This is a Duration object representing a length of time.
```
Comment (ishg-153) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:25]:
```suggestion
- `duration`: This is a Duration object representing a length of time.
```

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  - 'Methods'
+  - 'Data Structures'
+  - 'Date'
+  - 'Time'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.toDateTimePeriod()`** method in Kotlin is a versatile function that transforms a Duration object into a [DateTimePeriod](https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/-date-time-period/#:~:text=The%20date%20components%20are%3A%20years,periodUntil).) representation. This conversion is particularly useful when working with durations in Kotlin's date and time API, especially when expressing durations in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for easier interpretation and manipulation.
```
Comment (ishg-153) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:17]:
```suggestion
The **`.toDateTimePeriod()`** method in Kotlin is a versatile function that transforms a Duration object into a [DateTimePeriod](https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/-date-time-period/#:~:text=The%20date%20components%20are%3A%20years,periodUntil).) representation. This conversion is helpful when working with durations in Kotlin's date and time API, especially when expressing durations in terms of years, months, days, hours, minutes, seconds, and nanoseconds. The method normalizes the duration into a human-readable format for more effortless interpretation and manipulation.
```

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+
+## Syntax
+
+```pseudo
+duration.toDateTimePeriod(): DateTimePeriod
+```
+
+- **duration:** This is a Duration object representing a length of time.
+
+> **Note:** The method returns a DateTimePeriod object, which is a normalized representation of the input duration in terms of larger time units (years, months, days, etc.).
```
Comment (ishg-153) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:27]:
```suggestion
> **Note:** The method returns a DateTimePeriod object, a normalized representation of the input duration in terms of larger time units (years, months, days, etc.).
```

File: content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    val dateTimePeriod = totalWithDurationBreaks.toDateTimePeriod()
+
+    println("Total Duration in DateTimePeriod: $dateTimePeriod")
+}
+```
+
+The expected output should be similar to the following:
+
+```shell
+Total Duration in DateTimePeriod: DateTimePeriod(days = 4, hours = 18)
```
Comment (ishg-153) [inline content/kotlin/concepts/datetime/terms/toDateTimePeriod/toDateTimePeriod.md:57]: Can you confirm that this is the exact output you get on running the code example? I tried running it and got this:
`Total Duration in DateTimePeriod: P0Y0M1DT16H`

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@DylantheChicken I have suggested a few changes; please address them.

**Comment by caupolicandiaz** [comment]
@DylantheChicken @ishg-153 I made some changes to update the syntax and example to focus on the functionality with string objects since the date.time library is still considered experimental.

**Review by ishg-153 — APPROVED** [review]
@DylantheChicken @caupolicandiaz Okay great! This looks ready to merge.

**Pull request merged**